### PR TITLE
Resolve test case warnings

### PR DESCRIPTION
**Pull request by jamie-cunningham**
Resolve test case warnings. These mostly included divide by zero warnings and deprecations.

### CODE REVIEW

**Comment by jamie-cunningham** [comment]
Divide by zero cases were handled in place. Might be worth including a utility function for more robustness. Most cases were unit vector conversions. Open to discussion.

**Review by keatincf — CHANGES_REQUESTED** [review]
There's also some test warnings due to the name of some classes that start with Test and pytest is picking them up as tests rather

File: safe_autonomy_simulation/sims/inspection/inspection_points.py [hunk 1]
```diff
@@ -319,18 +321,25 @@ def _generate_points(
...
+                and np.linalg.norm(pos) != 0
+                else np.dot(-self.priority_vector, pos)
                 for pos in point_positions
             ]
         else:
             point_weights = []
             for pos in point_positions:
-                dot_prod = np.dot(-self.priority_vector, pos) / (np.linalg.norm(-self.priority_vector) * np.linalg.norm(pos))
+                dot_prod = np.dot(-self.priority_vector, pos) / (
+                    np.linalg.norm(-self.priority_vector) * np.linalg.norm(pos)
```
[the remark this answers is not in the document]
Comment (jamie-cunningham) [reply safe_autonomy_simulation/sims/inspection/inspection_points.py]: Added a manual check

**Comment by jamie-cunningham** [comment]
> There's also some test warnings due to the name of some classes that start with Test and pytest is picking them up as tests rather

Good catch, I missed those in my test output. I've renamed these classes and resolved the warnings.

**Review by keatincf — APPROVED** [review]

**Pull request merged**